config/connections.go

```diff
@@ -245,6 +270,17 @@ func CreateCloudConnection(
 		mosquittoConfig.Credentials.Password = settings.LocalPassword
 	}
 
+	switch u.Scheme {
```

Comment:
URL parsing and TLS check/initialization are duplicated code. Try to fix it in: CreateCloudConnection and CreateLocalConnection.